### DISCUSSION TITLE
Menu BackgroundColor SnoozeAlert

### DISCUSSION THREAD

**Discussion by JodokusQ** [NightscoutFoundation/xDrip#2954] [Ideas]
Is it possible to chance the backgroundcolor of the Menuitem SnoozeAlert (maybe the red from xDrip Icon).
2 reasons:
First of all, it is sometimes not so easy to read, if you are low (hypo).
Second for older people. If it alerts, you're lucky, if you have the the right glases (for reading) on your nose.

Thanks a lot for xDrip

**Comment by emp-00** [comment]
Great proposal, also for me this is an important menu item used multiple times every day and often I struggle hitting the right item. Another good alternative from my perspective: ==> make "Snooze Alert" directly available with an extra button e.g. here, option 1 and 2:

![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/c9cac11e-b7b4-4e4f-b7bf-59577241f007)

**Comment by Navid200** [comment]
There are times I would like to be able to snooze an alert as easily as possible.
An example is when I have just eaten something remedying a low and am driving.  But, an alert goes off.  I know that my glucose is coming up now because I have just eaten something, and I am driving.  So, I wish I could very easily snooze the alert.

On the other hands, there are times, I don't want to be able to easily snooze an alert.
An example of that is when I am sleep and a low alert goes off.  I am not all there; I have just opened my eyes.  If I can snooze the alert very easily, I may just go back to sleep.  That would be extremely dangerous.  What I really want is to wake up and go eat something.

Comparing those two scenarios, the first one is an inconvenience.  The second one is a nesessity.  

I don't think it is a very good idea to make it extremely easy to snooze an alert.

**Reply by emp-00** [reply]
Every human being has different preferences - therefore let's make this an option which is "OFF" by default but can be switched "ON" "by the educated user". This would be the best compromise and fits perfectly to the "xDrip+ experience". Power to the user.

**Comment by Navid200** [comment]
With respect to the background color of the snooze menu, let me see if I can find out how difficult, or easy, it may be to add the option.

**Reply by JodokusQ** [reply]
Thank you for your effort

**Reply by Navid200** [reply]
@JodokusQ 
Do you know that you can also snooze an alert by just swiping right on the xDrip notification associated with the alert?
So, you could snooze an alert without even opening xDrip itself.

**Reply by JodokusQ** [reply]
Hallo @Navid200,
I know that, but this is also difficult to read. Just if your quick enough to tap on the notification when it arrives. Another solution could be, to change the position. First or last menu entry, so you know where it is without reading.

**Reply by Navid200** [reply]
> Just if your quick enough to tap on the notification when it arrives

That's not true.  The notification remains in your notification area on the phone.  It does not disappear.
It's true that it disappears from the screen.  But, all you need to to do is to pull down from the top of the screen to reveal all the active notifications.

> Another solution ...  

I know.  I need everything to be known before we invest time into fixing something.  If you are not aware there are alternative ways for snoozing an alert, it would be jumping the gun to make any change.

**Reply by JodokusQ** [reply]
I know the possibilities. But reading the notifications to figure out the right notification is the same problem as reading the Menu.
If I wear my Smartwatch, I quit the alert with a Button.
Thats the easiest way. But most of the time, I don't.

**Reply by JodokusQ** [reply]
What about double tap (click) on widget?
Perhaps this is easy to implement. No impact on old Code.

**Reply by emp-00** [reply]
I also use the notification snooze function - BUT, this sznoozes for a rather short default time which I need/use during the day. My request for an extra button (or the menu color) is based on the fact, that I have a different snoozing strategy overnight. Here, I always snooze for e.g. 2 hours, thus much longer. This is because e.g. in case of a LOW alert, I eat/drink carbs - and since there are no more active IOB I can be 100% sure, that 2 hours are perfectly safe since the BG of course rises with the chosen dose. This is yearlong proven practical experience.

This is my main reason for requesting i.a. a button  --> because finding and reading the menu is especially hard at night.

So you can see, there are probably even more than these 2 use cases.

**Comment by Navid200** [comment]
I created a change to make the snooze item stand out and presented it to the developers.
The main objection was, what I had already told you in this same thread.  I am going to paste the comment exactly as it was presented to me for you to see.


> Do we need to make sure people are awake enough to silence the alert?

This is a serious concern that I had myself when I first saw your request.
If the only problem you experience for snoozing an alert is when you are half sleep, if we make it easier for you to snooze it, we will actually be doing you a disservice.

I have to say I am in agreement with the comment from the developers.
I have diabetes myself.  If I wake up in the middle of the night because of an alert and can very easily snooze the alert, I may just go back to sleep instead of getting out of bed and treating what caused the alert to go off in the first place.  So, as an xDrip user, I don't want the snooze to become any easier than it already is.

I am sorry.  I hope you don't think this is just an argument.  I will go to bat for you if you can convince me.

**Reply by emp-00** [reply]
In principle, I do of course understand your hestitation - BUT, let's face the facts ;-) Honestly, guys:

- xDrip+ is distributed with this disclaimer: "Use xDrip+ at your own risk, and do not use the information or code to make medical decisions.": Every user accepts this when using xDrip+
- This means it is per design a tool for "educated diabetics": I just give you the following concrete example for comparison
- xDrip+ already allows changing the default snooze time to 12 hours! How can you guys then argue, that it's "unsafe" just changing the color of the snooze menu option or adding a button which allows for easier access of the snooze functionality?

Every user already accepts "risks" when using xDrip+ and this is excatly the great value of xDrip: it can be configured much better to every diabetics needs compared to all commercial apps. As a matter of fact, every diabetic has different needs and that's also why I love xDrip+. It saves time, gives me much more freedom and it clearly improves my diabetes statistics.

Concrete proposal was: Give the educated user the choice to switch ON the **option** to change the color of the Snooze menu, default = OFF. One of many other great options. This is much less "dangerous" compared to setting the default snooze time to 12 hours, right? Honestly! :-)

Please don't take this as personal critics -> I fully understand your hesitation but maybe my argumentation helps to convince the developers? Keeping my fingers crossed and thank you so much, @Navid200, for your help. Highly appreciated!

**Reply by Navid200** [reply]
No, it's not personal.  I asked for these discussions to be opened because when used properly, as we are, they can help us learn and grow.

If I set my snooze time to 12 hours, I will still have to wake up to snooze it before I go back to sleep.

If I make snoozing the alert easier, the sleepy Navid will be happy because he can snooze and go right back to sleep.  But, the awake Navid the next morning could be seriously mad.

So, I don't consider setting the snooze time to 12 hours as dangerous as making snoozing easier.

I still don't have an explanation why anyone needs to snooze an alert before fully waking up.
I know you say just add it and let those who want it use it and those who don't want it leave it alone.

We can't move forward that way.
xDrip already has too many settings.  We cannot keep adding more settings without asking questions.

My question is why anyone needs to snooze an alert before fully waking up?
If you have set an alert, it is meant to wake you.  If you don't want to wake up, you can disable the alert.

**Comment by emp-00** [comment]
The answer is my way of living / my circumstances, already explained above, more details happily given here:

1. I don't have a pump, no loop, just plain-and-simple pen insulin injection
2. I often eat dinner e.g. 1 hour before going to sleep (life circumstances), thus having still some active COB / IOB
3. So it can happen I do get a HIGH or LOW alert e.g. 1 hour after going to sleep
4. Now, being AWAKE thanks to the alarm, of course I act accordingly: eating carbs or injecting insulin
5. however, now I have the need for a delayed snooze in this case: Every diabetic knows it takes >1 hours until the insulin reaches peak effectivity (3 hours total) and carbs also take similar time until they raise your BG and everything equilibrates again
6. Important: as an educated xDrip+ diabetic, I know based on yearlong experience, I can snooze for e.g. 2 hours in such a case (avoiding unesccessary follow-up alarms) but it's simply not easy to find the darn snooze menu option in such a dark nightly situation...)
7. Another good reason for the need activating a dedicated snooze time at night: GGM sensors very often are extremely unreliable during the first 24 hours and if I get an alarm due to this situation in the middle of the night, I wanna find the darn menu fast :-)

To put it shorty; I just want to improve my quality of sleep by having the ability to find and open the snooze menu at night - I really often hit the menu items above and below in such cases. This is really annoying. During daytime I have no issues but at night, it's the only thing that bothers me using xDrip+.

Honestly, I don't understand your argument "_I don't consider setting the snooze time to 12 hours as dangerous as making snoozing easier_" - this is not about "making snoozing easier", it's about "making it easier to hit the correct menu item at night". On the other hand, setting a "12 hours" snooze time, I personally would consider as a really dangerous situation in "normal" circumstances - and this is already possible with xDrip+. The latter is perfectly OK, please don't remove this functionality - again, it's all about "users choice". We all have different ways of living :-) Thanks again for your understanding and considering this small change.

**Comment by Navid200** [comment]
In order to snooze the alert using the snooze menu, you will need to unlock your phone first and then, be in the xDrip app.

If the phone is unlocked and you are in xDrip, there is an easier way to snooze the alert.  Under those conditions, by default, you can just press either of the volume buttons on the phone to snooze the alert.
Wouldn't that accomplish the same objective you are after?

**Reply by JodokusQ** [reply]
Hello Navid,
I wasn't interested in changing the functionality of the alert snooze. It's good the way it is.
I just want better manageability.

Logically, the alarm comes unexpectedly, i.e. unprepared.

That means no reading glasses on your nose.

In the case of a hypo, it is generally not that easy to read (colored spots in front of the eyes).

Therefore, my wish was to achieve the existing functionality in a slightly simpler way.

I'm sure I'm not the only one struggling with Altersweitsichtigkeit (presbyopia?).

I didn't even want to have the discussion as to whether xDrip has an educational mandate.

In my opinion, software should provide opportunities. Everyone has to know for themselves what is used.

Best regards Jodokus 

I will try the Volume-Buttons, maybe it helps.

**Reply by Navid200** [reply]
@JodokusQ You can snooze by pressing the volume buttons.  Isn't that easier than looking for the right item in the menu?

**Reply by emp-00** [reply]
@Navid200 : No, this just does not solve my issue - see my description(s). When using the volume buttons only the default snooze time (20 mins) is activated, but my intention is to set a situation-specific (IOB/COB dependent) **longer** snooze time of e.g. 1 or 2 hours in order to avoid unnecessary follow-up alarms. Therefore I have to use the **snooze menu** which is just really hard to find at night (and with bad eyesight, just as @JodokusQ ).

One more very, very simple and concrete suggestion: Please just add e.g. *** (three asterisks) as text after "Snooze Alert" (see screenshot). In the xDrip code this should just requires changing the string "_Snooze Alert_" to "_Snooze Alert  ***_". Super-simple change but still great improvement, because this more clearly distinguishes the menu items. Please note, that they all have almost the same text length, that's why it's so hard for us to hit the correct item at night and without glasses. Check the screenshot and try imagine you can't see 100% clearly... No, not because of a dangerous hypo but just because of bad eyesight + darkness. Would this be a workable compromise?

RIGHT SIDE - a simulation of how things look at night for me and @JodokusQ and many others, I guess (*** = improvement)
![image](https://github.com/NightscoutFoundation/xDrip/assets/60714349/cbc4a3df-1daf-46d9-8c49-4df56ae418a7)

**Comment by Navid200** [comment]
@emp-00 I have wanted to address the concerns of the individual who started this thread.
But, you keep interjecting.
This is a discussion.  You are welcome to add your opinion.  But, the intent of this thread is to address the concern of the person who started the thread.

If you have detail needs and requirements, please start a new thread to avoid possibly taking over an existing one started by someone else.

Every Glucose level alert has its own snooze time as a parameter.  You can define the snooze time for each one of your glucose alerts separately.  Then, when you use the volume button to snooze, the alert will be snoozed for the specified snooze time by you.

**Reply by emp-00** [reply]
I just made a concrete proposal which could be a solution as well for the person who started the thread. It was not my intention to interject, I apologize for giving this impression. For me we're talking about the same request.

**Reply by Navid200** [reply]
@JodokusQ mentioned that he/she does not want to have to look for their glasses when they wake up in the middle of the night in order to snooze the alert.
I have suggested they can  do that by just pressing the volume button and am waiting for their response.

You have brought up a good point that snoozing from the snooze menu allows you to snooze for a specific length of time.
This may or may not concern JodokusQ.
Regardless, it is very important.  So, even if they don't care about it, you have a valid point.

I thank you for bringing that up.
Would you please tell me if you can specify the length of time you want to snooze your alert using the alert edit page?

**Reply by emp-00** [reply]
Thanks Navid. My issue is the difficulty hitting the correct menu item in the middle of the night without glasses just as @JodokusQ explained as well. He proposed a different menu color and I made a more simple proposal (3 asterisks) since you're very hesitant to implement this change. I just want to be constructive in order to find a compromise solution. 

I explicitly don't want to set a fixed snooze time but I want to be able to use the existing snooze time selection wheel. Usually I choose 60, 90 or 120 minutes - strongly depending on my IOB/COB... So in my case, only making the "Snooze Alert" menu item stand out from "System Status" and "Note Search" would be a simple and great improvement.

**Comment by JodokusQ** [comment]
The volume buttons work. This actually helps if the volume of the mobile phone is also adjusted. But the point of emp-00 is not entirely unjustified. If you want to change the time, you have to go into the menu.

**Reply by Navid200** [reply]
When an alert is playing, if your screen is unlocked and you are in xDrip, gently pressing a volume button snoozes the alert with no impact on the phone volume.

I have communicated with the developers again with a request to make the snooze alert button stand out, and we have to be patient.

**Reply by JodokusQ** [reply]
Thanks Navid :-)
